pulsar-client-cpp/lib/ConsumerImpl.cc

```diff
@@ -686,4 +687,44 @@ int ConsumerImpl::getNumOfPrefetchedMessages() const {
     return incomingMessages_.size();
 }
 
+Result ConsumerImpl::getConsumerStats(BrokerSideConsumerStats& brokerSideConsumerStats, int partitionIndex) {
+    if (partitionIndex != -1) {
+        LOG_WARN(getName() << " In getConsumerStats call - Ignoring the partitionIndex since the topic is not partitioned")
+    }
+
+    if (!isOpen()) {
+        LOG_ERROR(getName() << " In getConsumerStats call - Client connection is not open, please try again later.")
+        return ResultConsumerNotInitialized;
+    }
+
+    if (brokerSideConsumerStats_.isValid()) {
+        LOG_INFO(getName() << " In getConsumerStats call - Serving data from cache");
+        brokerSideConsumerStats = brokerSideConsumerStats_;
+        return ResultOk;
+    }
+
+
+    ClientConnectionPtr cnx = getCnx().lock();
+    if (cnx) {
+        if (cnx->getServerProtocolVersion() >= proto::v7) {
+            ClientImplPtr client = client_.lock();
+            uint64_t requestId = client->newRequestId();
+            LOG_DEBUG(getName() <<
+                    " Sending getConsumerStats command for Consumer - " << getConsumerId() << ", requestId - "<<requestId);
+
+            BrokerSideConsumerStats consumerStats;
+            Result res = cnx->newConsumerStats(topic_, subscription_, consumerId_, requestId).get(consumerStats);
```

Comment:
Do you think Is it worth it to provide callback option also in case of client doesn't want to block such as: sendAsync(msg, callBack)